src/test/java/uk/gov/ch/developer/docs/controller/developer/UserCallbackControllerTest.java

```diff
@@ -1,49 +1,78 @@
 package uk.gov.ch.developer.docs.controller.developer;
 
-import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
 import static org.mockito.Mockito.when;
 
-import com.nimbusds.jose.Payload;
-import javax.servlet.http.HttpServletRequest;
-import net.minidev.json.JSONObject;
+import java.io.IOException;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.TestInstance;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.InjectMocks;
 import org.mockito.Mock;
+import org.mockito.Spy;
 import org.mockito.junit.jupiter.MockitoExtension;
-import uk.gov.ch.oauth.IOauth;
-import uk.gov.ch.oauth.identity.IIdentityProvider;
+import uk.gov.ch.oauth.Oauth2;
+import uk.gov.ch.oauth.identity.IdentityProvider;
 
 @ExtendWith(MockitoExtension.class)
-@TestInstance(TestInstance.Lifecycle.PER_CLASS)
 public class UserCallbackControllerTest {
 
+    public static final String HTTP_EXAMPLE_COM_REDIRECT = "http://example.com/redirect";
     @Mock
-    private IOauth oauth;
+    IdentityProvider identityProvider;
     @Mock
-    private HttpServletRequest httpServletRequest;
+    private HttpServletResponse servletResponse;
     @Mock
-    private Payload payload;
-    @Mock
-    private JSONObject jsonObject;
-    @Mock
-    private IIdentityProvider identityProvider;
+    private Oauth2 oauth2;
 
     @InjectMocks
+    @Spy
```

Comment:
We probably don't need this, doesn't look like we are monitoring interactions on this object.

Reply:
Agreed, I have removed the `@Spy`